velero-plugins/imagestream/shared_test.go

```diff
@@ -0,0 +1,162 @@
+package imagestream
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/konveyor/openshift-velero-plugin/velero-plugins/clients"
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/restmapper"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+func TestGetRegistryEnvsForLocation(t *testing.T) {
+	type args struct {
+		location  string
+		namespace string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		objs    []metav1.Object
+		want    []string
+		wantErr bool
+	}{
+		{
+			name: "simple aws case, operator created registry secret",
+			args: args{
+				location: "bsl1",
+				namespace: "ns1",
+			},
+			objs: []metav1.Object{
+				&corev1.Namespace{
+					TypeMeta: metav1.TypeMeta{
+						Kind: "Namespace",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "ns1",
+					},
+				},
+				&corev1.Secret{
+					TypeMeta: metav1.TypeMeta{
+						Kind: "Secret",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "oadp-bsl1-aws-registry-secret",
+						Namespace: "ns1",
+					},
+					Data: map[string][]byte{
+						"access_key": []byte("ak"),
+						"secret_key": []byte("sk"),
+					},
+				},
+				&velerov1.BackupStorageLocation{
+					TypeMeta: metav1.TypeMeta{
+						Kind: "BackupStorageLocation",
+						APIVersion: velerov1.SchemeGroupVersion.String(),
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "bsl1",
+						Namespace: "ns1",
+					},
+					Spec: velerov1.BackupStorageLocationSpec{
+						Provider: "aws",
+						Default: true,
+						StorageType: velerov1.StorageType{
+							ObjectStorage: &velerov1.ObjectStorageLocation{
+								Bucket: "buc",
+								Prefix: "prefix",
+							},
+						},
+						Credential: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: "cloud-credentials",
+							},
+							Key: "cloud",
+						},
+					},
+				},
+			},
+		},
+	}
+	scheme := runtime.NewScheme()
+	corev1.AddToScheme(scheme)
+	metav1.AddMetaToScheme(scheme)
+	velerov1.AddToScheme(scheme)
+	testEnv := &envtest.Environment{
+		Scheme: scheme,
+	}
+	cfg, err := testEnv.Start()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer testEnv.Stop()
+	clients.SetInClusterConfig(cfg)
+	client, err := dynamic.NewForConfig(cfg)
+	if err != nil {
+		panic(err)
+	}
+	// corev1client, err := corev1c.NewForConfig(cfg)
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(cfg)
+	if err != nil {
+		panic(err)
+	}
+	// GR
+	groupResources, err := restmapper.GetAPIGroupResources(discoveryClient)
+	if err != nil {
+		panic(err)
+	}
+	// GVR
+	rm := restmapper.NewDiscoveryRESTMapper(groupResources)
```

Comment:
Tried to use dynamic client in unit test.. maybe bad idea.. does anyone know how to fix this?
```
panic: no matches for kind "BackupStorageLocation" in version "velero.io/v1" [recovered]
	panic: no matches for kind "BackupStorageLocation" in version "velero.io/v1"
```